src/yellowpaper/chapters/03-implementation/data-disclosure.tex

```diff
@@ -68,14 +87,34 @@ \subsection{Data Disclosure}
     \hline
     \textbf{Condition} & \textbf{Step} & \textbf{Consequence}\\ \hline
     \endhead
-    $\Delta t > \tau$ & 2 & \multirow{4}{*}{
+
+    $\Delta t > \tau$ & \ref{step:seller-deposits-money} & \multirow{4}{*}{
       \parbox{\linewidth}{\tParty{B}, \tParty{E} get their money back because \tParty{E}~wasn't able to correctly transfer the data to~\tParty{B}.}
     } \\ \cline{1-2}
-    $\Delta t > \tau$ & 4 & \\ \cline{1-2}
-    $R' \neq R$ & 4 & \\ \cline{1-2}
-    $s' \neq s$ & 4 & \\ \hline
-    $\Delta t > \tau$ & 5 & \tParty{B}, \tParty{E} get their money back because \tParty{B}~has received the encrypted data, but \tParty{E}~nas not been able to share the key $k$ for it \\ \hline
-    $\Delta t > \tau$ & 6 & \tParty{E} gets $C_E$ and $C_D$: \tParty{E} correctly shared data to \tParty{B} \\ \hline
+    $\Delta t > \tau$ & \ref{step:buyers-receipt} & \\ \cline{1-2}
+    $R' \neq R$ & \ref{step:buyers-receipt} & \\ \cline{1-2}
+    $s' \neq s$ & \ref{step:buyers-receipt} & \\ \hline
+    $\Delta t > \tau$ & \ref{step:secret-sharing} & \tParty{B}, \tParty{E} get their money back because \tParty{B}~has received the encrypted data, but \tParty{E}~nas not been able to share the key $k$ for it \\ \hline
+    $\Delta t > \tau$ & \ref{step:arbitrage} & \tParty{E} gets $C_E$ and $C_D$: \tParty{E} correctly shared data to \tParty{B} \\ \hline
     \multicolumn{2}{|c|}{Protocol finishes} & The dispute situation. In case \tParty{B} proofs \tParty{E} cheated, \tParty{E} loses her security deposit $C_E$. Otherwise, \tParty{E} receives both $C_E$ and $C_D$. \\ \hline
   \end{longtabu}
 \end{table}
+
+The proposed algorithm (though with some modifications) can be applied to object disclosure and personalization requests. Here we define the modifications of the protocol so that it can support these request types:
+\begin{enumerate}
+  \item Object disclosure
+    \begin{itemize}
+      \item $Q: \Root(\MerkleTree(\mathit{Object}))$ -- query by the object hash.
+      \item $D: \mathit{Object}$ -- the data being revealed is an \textit{object:} uncategorized blob of data relevant to a particular transaction.
+      \item Validation: check the Merkle root of the object. In case the Merkle root does not match, the buyer sends a chunk with proofs to the contract.
+      \item In case of the successful trade the student receives a part of money.
+    \end{itemize}
+
+  \item Personal data disclosure
+    \begin{itemize}
+      \item $Q: \PubKey{S}$ -- query by the public key of a student.
+      \item $D: \Sign_\Party{S}(\mathit{personalInfo})$ -- the data being revealed is the personal information signed by the student.
+      \item Validation: the buyer checks the student's signature. In case of a dispute the contract checks the signature.
+      \item In case of the successful trade the student receives a part of money.
```

Comment:
Same here.

Reply:
How the "student's signature" can be checked by contract and isn't the personal data the thing to be checked?